### PR TITLE
fix: merge approval feedback into tool result instead of pushing duplicate (ROO-410)

### PR DESCRIPTION
**Pull request by daniel-lxs**
## Problem

When a user approves a tool execution AND provides feedback text, `askApproval` in `presentAssistantMessage.ts` was pushing the feedback as a separate `tool_result`. This created duplicate `tool_result` blocks with the same `tool_use_id` because the actual tool also pushes its result when it executes.

This is a protocol violation for the Anthropic API and was documented as GitHub #10465. A workaround existed in `validateAndFixToolResultIds.ts` that deduplicated results AFTER they were created, but the root cause was never addressed.

## Solution

Instead of pushing approval feedback as a separate tool_result, the feedback is now **stored** and **merged** into the actual tool's result when `pushToolResult` is called:

1. Added `approvalFeedback` storage variable to hold feedback until tool completes
2. Modified `pushToolResult` to merge stored feedback into the tool's actual result content
3. Updated `askApproval` to store feedback instead of pushing it as a separate tool_result
4. This applies to both MCP tools and regular tools, and both native and XML protocols

## Changes

- `src/core/assistant-message/presentAssistantMessage.ts` - root fix for both MCP and regular tools
- `src/core/task/validateToolResultIds.ts` - updated comments (deduplication kept as safety net)

## Testing

- All existing tests pass (5153 tests)
- The deduplication safety net in `validateAndFixToolResultIds.ts` remains as defense-in-depth

Closes ROO-410

----

> [!IMPORTANT]
> Fixes duplicate `tool_result` issue by merging approval feedback into tool results in `presentAssistantMessage.ts`.
> 
>   - **Behavior**:
>     - `presentAssistantMessage.ts`: Merges approval feedback into tool results instead of creating duplicate `tool_result` blocks.
>     - Applies to both MCP tools and regular tools, and both native and XML protocols.
>   - **Implementation**:
>     - Adds `approvalFeedback` variable to store feedback until tool completion.
>     - Modifies `pushToolResult` to merge stored feedback into tool results.
>     - Updates `askApproval` to store feedback instead of pushing it as a separate `tool_result`.
>   - **Validation**:
>     - `validateToolResultIds.ts`: Keeps deduplication as a safety net for unknown edge cases.
>   - **Testing**:
>     - All existing tests pass (5153 tests).
>     - Deduplication remains as defense-in-depth.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for e4e67e27dc9c8b1e8f70a0bbd56d70ebcbf887f3. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task on Roo Cloud](https://app.roocode.com/cloud-jobs/0daZGtOp?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Reviewed the changes to fix duplicate tool_result blocks. Found one issue that needs attention:

- [ ] XML protocol path doesn't handle approval feedback images (approvalFeedback.images is ignored while native protocol handles it correctly)

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=10519&commentId=3719833402).</sub>

File: src/core/assistant-message/presentAssistantMessage.ts [hunk 1]
```diff
@@ -544,15 +578,44 @@ export async function presentAssistantMessage(cline: Task) {
...
+					}
+
+					// Merge approval feedback into tool result (GitHub #10465)
+					if (approvalFeedback) {
+						const feedbackText = formatResponse.toolApprovedWithFeedback(
+							approvalFeedback.text,
+							toolProtocol,
+						)
+						resultContent = `${feedbackText}\n\n${resultContent}`
+					}
```
[the remark this answers is not in the document]
Comment (cte) [reply src/core/assistant-message/presentAssistantMessage.ts:599]: Is this legit?

[the remark this answers is not in the document]
Comment (daniel-lxs) [reply src/core/assistant-message/presentAssistantMessage.ts:599]: You shouldn't be able to create XML tasks anymore anyways so I'm not sure if it's worth fixing this at all
